node/network/availability-recovery/src/lib.rs

```diff
@@ -564,55 +565,22 @@ async fn handle_from_interaction(
 			state.availability_lru.put(candidate_hash, result);
 		}
 		FromInteraction::MakeRequest(id, candidate_hash, validator_index, response) => {
-			let relay_parent = state.live_block_hash;
+			let (tx, rx) = mpsc::channel(2);
 
-			let session_index = match state.interactions.get(&candidate_hash) {
-			    Some(interaction_handle) => interaction_handle.session_index,
-			    None => {
-					tracing::warn!(
-						target: LOG_TARGET,
-						"Interaction under candidate hash {} is missing",
-						candidate_hash,
-					);
-
-					return Ok(());
-				}
+			let message = NetworkBridgeMessage::ConnectToValidators {
+				validator_ids: vec![id.clone()],
+				connected: tx,
 			};
 
-			// Take the validators we are already connecting to and merge them with the
-			// newly requested one to create a new connection request.
-			let new_discovering_validators_set = state.discovering_validators.keys()
-				.cloned()
-				.chain(std::iter::once(id.clone()))
-				.collect();
+			ctx.send_message(AllMessages::NetworkBridge(message)).await;
 
-			// Issue a NetworkBridgeMessage::ConnectToValidators.
-			// Add the stream of connected validator events to state.connecting_validators.
-			match validator_discovery::connect_to_past_session_validators(
-				ctx,
-				relay_parent,
-				new_discovering_validators_set,
-				session_index,
-			).await {
-				Ok(new_connection_request) => {
-					state.connection_requests.put(relay_parent, new_connection_request);
-					// Add an AwaitedChunk to the discovering_validators map under discovery_pub.
-					let awaited_chunk = AwaitedChunk {
-						validator_index,
-						candidate_hash,
-						response,
-					};
-
-					state.discovering_validators.entry(id).or_default().push(awaited_chunk);
-				}
-				Err(e) => {
-					tracing::debug!(
-						target: LOG_TARGET,
-						err = ?e,
-						"Failed to create a validator connection request",
-					);
-				}
-			}
+			state.discovering_validators.entry(id).or_default().push(AwaitedChunk {
+				validator_index,
+				candidate_hash,
+				response,
+			});
+
+			state.connecting_validators.push(rx);
```

Comment:
With this `StreamUnordered` I do wonder about when the `rx` is actually dropped as that is how the network bridge is actually informed.